### ISSUE TITLE
[facebook:reel] ERROR No video formats found! when downloading Facebook Reels

### ISSUE THREAD

**Issue by ekiara** [yt-dlp/yt-dlp#7469]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.06.22** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Kenya

### Provide a description that is worded well enough to be understood

After viewing a Facebook reel on the website (the URL format is as follows:  https://www.facebook.com/reel/682865820350105) I attempted to download the video using `yt-dlp` however I got the folloing error response:

```zsh

$ ./yt-dlp "https://www.facebook.com/reel/682865820350105"
[facebook:reel] Extracting URL: https://www.facebook.com/reel/682865820350105
[facebook] Extracting URL: https://m.facebook.com/watch/?v=682865820350105&_rdr
[facebook] 682865820350105: Downloading webpage
ERROR: [facebook] 682865820350105: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U

$ ./yt-dlp --version
2023.06.22

```

Also I attempted to view the reel again but in an Incognito Tab to confirm that the video is public and it worked: i.e. I could view the video, meaning it is a public URL not needed Facebook credentials to view.

NOTE: Using the command line utility, and NOT using the API.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.facebook.com/reel/682865820350105']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.06.22 [812cdfa06] (zip)
[debug] Python 3.10.7 (CPython x86_64 64bit) - Linux-5.19.0-43-generic-x86_64-with-glibc2.36 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.1 (setts), ffprobe 5.1.1
[debug] Optional libraries: certifi-2022.12.07, secretstorage-3.3.3, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1851 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.06.22, Current version: stable@2023.06.22
Current Build Hash: 924464eb831bdb99d854b9d47ff9e1c0870eab276113adaf8a74a8561ad332b9
yt-dlp is up to date (stable@2023.06.22)
[facebook:reel] Extracting URL: https://www.facebook.com/reel/682865820350105
[facebook] Extracting URL: https://m.facebook.com/watch/?v=682865820350105&_rdr
[facebook] 682865820350105: Downloading webpage
ERROR: [facebook] 682865820350105: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the a
ppropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/YoutubeDL.py", line 1555, in wrapper
    return func(self, *args, **kwargs)
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/YoutubeDL.py", line 1652, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/YoutubeDL.py", line 1711, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/YoutubeDL.py", line 2653, in process_video_result
    self.raise_no_formats(info_dict)
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/YoutubeDL.py", line 1067, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [facebook] 682865820350105: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?
q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

**Comment by bashonly** [comment]
This issue seems to occur with all reels URLs now, including the extractor test. I don't think redirecting to the mobile `/watch/` page and handing it off to `FacebookIE` will work anymore.

**Comment by october262** [comment]
for this test URL - https://www.facebook.com/reel/682865820350105
paste the URL in a new tab. change www to mbasic, press enter and then 
copy the URL to download the video.

**Comment by ekiara** [comment]
> for this test URL - https://www.facebook.com/reel/682865820350105 paste the URL in a new tab, change www to mbasic, press enter and then copy the URL to download the video.

@october262 Okay so replacing the `www` with `mbasic` generates a URL that does will not work in an Incognito Tab because the browser redirects to a Facebook login page.

And also with `yt-dlp` and using a `cookies.txt` file with the URL generated by replacing `www` with `mbasic` does not work at the moment:

```zsh
$  ./yt-dlp -vU --cookies ./cookies.txt "https://mbasic.facebook.com/watch/?extid=reels&v=682865820350105&paipv=0&eav=AfYtf1pZlkowA3bOaCUPMfJ1S_Cl6AttcatLUIKTgT2n3ZKEVCdS-Oc4dQ3YgEhluN0&_rdr"
[debug] Command-line config: ['-vU', '--cookies', './cookies.txt', 'https://mbasic.facebook.com/watch/?extid=reels&v=682865820350105&paipv=0&eav=AfYtf1pZlkowA3bOaCUPMfJ1S_Cl6AttcatLUIKTgT2n3ZKEVCdS-Oc4dQ3YgEhluN0&_rdr']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.06.22 [812cdfa06] (zip)
[debug] Python 3.10.7 (CPython x86_64 64bit) - Linux-5.19.0-43-generic-x86_64-with-glibc2.36 (OpenSSL 3.0.5 5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.1 (setts), ffprobe 5.1.1
[debug] Optional libraries: certifi-2022.12.07, secretstorage-3.3.3, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1851 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.06.22, Current version: stable@2023.06.22
Current Build Hash: 924464eb831bdb99d854b9d47ff9e1c0870eab276113adaf8a74a8561ad332b9
yt-dlp is up to date (stable@2023.06.22)
[facebook] Extracting URL: https://mbasic.facebook.com/watch/?extid=reels&v=682865820350105&paipv=0&eav=AfYtf1pZlkowA3bOaCUPMfJ1S_Cl6AttcatLUIKTgT2n3ZKEVCdS-Oc4dQ3YgEhluN0&_rdr
[facebook] 682865820350105: Downloading webpage
[facebook] 682865820350105: Downloading webpage
ERROR: [facebook] 682865820350105: Cannot parse data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/extractor/common.py", line 708, in extract
    ie_result = self._real_extract(url)
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/extractor/facebook.py", line 707, in _real_extract
    return self._extract_from_url(real_url, video_id)
  File "/home/emk/Developer/SCRATCH/youtube-downloader/./yt-dlp/yt_dlp/extractor/facebook.py", line 654, in _extract_from_url
    raise ExtractorError('Cannot parse data')
```

As @bashonly says, it looks like this issue is affecting all reels URLs now.

**Comment by october262** [comment]
download works ok for me either with yt-dlp or by right click & save.
i take it your not signed in to facebook correct ??

**Comment by cunlem** [comment]
Just in case: an alternative page for watching a reel is 
https://www.facebook.com/photo.php?fbid=682865820350105
It shows reel video in more regular layout which might be easier to scrape

**Comment by arfathyahiya** [comment]
can't we download the reels using requests and bs4 package?
from this url
https://m.facebook.com/watch/?v=682865820350105
 The issue I found using this endpoint is that only lower quality video is available

**Comment by Arsh1219** [comment]
I am waiting for some workaround. Has anyone found it yet?

**Comment by andreabravetti** [comment]
I'm having the same error with [https://www.facebook.com/reel/772744577921748](https://www.facebook.com/reel/772744577921748)

**Comment by vjaideep08** [comment]
How to download or extract the meta-data of facebook reels using yt-dlp with python. 
can someone please help me!

**Comment by vjaideep08** [comment]
@bashonly please show the correct solution for the above issue

**Comment by bashonly** [comment]
@vjaideep08 see the linked pull request, #7564

**Comment by someziggyman** [comment]
Hey guys, looks like the issue is back again https://www.facebook.com/reel/573336878299071 or https://www.facebook.com/watch/?v=573336878299071 both return ERROR: [facebook] 573336878299071: Cannot parse data

Log below:

./yt-dlp -Uv -F "https://www.facebook.com/reel/573336878299071"
[debug] Command-line config: ['-Uv', '-F', 'https://www.facebook.com/reel/573336878299071']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Python 3.11.4 (CPython arm64 64bit) - macOS-13.5-arm64-arm-64bit (OpenSSL 3.1.2 1 Aug 2023)
[debug] exe versions: phantomjs 2.1.1
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Extractor Plugins: SamplePluginIE
[debug] Post-Processor Plugins: SamplePluginPP
[debug] Loaded 1937 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: 53030e66cc63e31cd608efb9befc5f31cf2f089ea80388189f36b9f2de7d33a2
yt-dlp is up to date (stable@2023.07.06)
[facebook:reel] Extracting URL: https://www.facebook.com/reel/573336878299071
[facebook] Extracting URL: https://m.facebook.com/watch/?v=573336878299071&_rdr
[facebook] 573336878299071: Downloading webpage
[facebook] 573336878299071: Downloading webpage
ERROR: [facebook] 573336878299071: Cannot parse data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "./yt-dlp/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/facebook.py", line 733, in _real_extract
    return self._extract_from_url(real_url, video_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "./yt-dlp/yt_dlp/extractor/facebook.py", line 680, in _extract_from_url
    raise ExtractorError('Cannot parse data')

**Comment by bashonly** [comment]
#4311

FB extractor is currently compeletely broken
